File: app/lib/domain/entities/trade_journal.dart
/// File: app/lib/domain/entities/trade_journal.dart
///
/// Role:
///   Dart mirror of `shared.domain.trade_journal.TradeJournalEntry`.
///
/// Fields:
///   - id, tradeId, direction
///   - reason (REQUIRED)
///   - appliedPrincipleIds (List<int>)
///   - postReview (nullable)
///   - createdAt
///
/// UI workflow:
///   - The Portfolio screen, on order success, opens an "EntryForm"
///     modal that POSTs this entity. `reason` is validated client-side
///     before allowing submit.
///   - The Trade Journal screen lists entries with joined Trade + Principles.
///
/// Connected modules:
///   - data/api/trade_journal_api.dart
///   - presentation/trade_journal/
///   - presentation/portfolio/ (modal trigger)

File: shared/domain/trade_journal.py
"""Trade journal domain models."""

from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel

from shared.domain.trade import TradeDirection


class TradeJournalEntry(BaseModel):
    id: int
    trade_id: int
    direction: TradeDirection
    reason: str
    applied_principle_ids: list[int]
    post_review: str | None
    created_at: datetime
